test/CMakeLists.txt

```diff
@@ -68,7 +46,7 @@ nextcloud_add_test(Permissions "")
 nextcloud_add_test(SelectiveSync "")
 nextcloud_add_test(DatabaseError "")
 nextcloud_add_test(LockedFiles "../src/gui/lockwatcher.cpp")
-nextcloud_add_test(FolderWatcher "${FolderWatcher_SRC}")
+nextcloud_add_test(FolderWatcher "")
```

Comment:
We can probably remove that second parameter to nextcloud_add_test now, right? (to be done in a separate commit of this PR).

Reply:
I think we need it sometimes for extra dependencies like pushnotificationstestutils and so on unless we make these helper utils a library like syncenginetestutils and link it with all tests in `nextcloud_add_test`. But this might be some unnecessary linking which costs compile time (if we get a lot of tests). Maybe the best way would be to refactor `nextcloud_add_test` so that you can give it additional libraries that you want to be linked with the test.

Reply:
Well, pushnotificationstestutils and syncenginetestutils could be in the same library for a start? Having a single startutils library would be useful.